// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SurefireParser can parse a Surefire v2 report for JUnit 4 1`] = `
SurefireReport {
  "errors": 2,
  "failures": 4,
  "name": "org.example.JUnit4Test",
  "skipped": 4,
  "testCases": Array [
    Object {
      "className": "org.example.JUnit4Test",
      "message": "Assertion message expected:<5> but was:<4>",
      "result": "failure",
      "stackTrace": "java.lang.AssertionError: Assertion message expected:<5> but was:<4>
	at org.example.JUnit4Test.testAssertionFailureWithMessage(JUnit4Test.java:26)",
      "testName": "testAssertionFailureWithMessage",
    },
    Object {
      "className": "org.example.JUnit4Test",
      "message": "",
      "result": "error",
      "stackTrace": "java.lang.RuntimeException
	at org.example.JUnit4Test.testError(JUnit4Test.java:41)",
      "testName": "testError",
    },
    Object {
      "className": "org.example.JUnit4Test",
      "message": "",
      "result": "failure",
      "stackTrace": "java.lang.AssertionError
	at org.example.JUnit4Test.testFail(JUnit4Test.java:31)",
      "testName": "testFail",
    },
    Object {
      "className": "org.example.JUnit4Test",
      "message": "Error message",
      "result": "error",
      "stackTrace": "java.lang.RuntimeException: Error message
	at org.example.JUnit4Test.testErrorWithMessage(JUnit4Test.java:46)",
      "testName": "testErrorWithMessage",
    },
    Object {
      "className": "org.example.JUnit4Test",
      "message": "Ignore message",
      "result": "skipped",
      "stackTrace": undefined,
      "testName": "testIgnoredWithMessage",
    },
    Object {
      "className": "org.example.JUnit4Test",
      "message": "Assumption message",
      "result": "skipped",
      "stackTrace": "org.junit.AssumptionViolatedException: Assumption message
	at org.example.JUnit4Test.testSkippedWithMessage(JUnit4Test.java:56)",
      "testName": "testSkippedWithMessage",
    },
    Object {
      "className": "org.example.JUnit4Test",
      "message": "Failure message",
      "result": "failure",
      "stackTrace": "java.lang.AssertionError: Failure message
	at org.example.JUnit4Test.testFailWithMessage(JUnit4Test.java:36)",
      "testName": "testFailWithMessage",
    },
    Object {
      "className": "org.example.JUnit4Test",
      "message": "expected:<5> but was:<4>",
      "result": "failure",
      "stackTrace": "java.lang.AssertionError: expected:<5> but was:<4>
	at org.example.JUnit4Test.testAssertionFailure(JUnit4Test.java:20)",
      "testName": "testAssertionFailure",
    },
    Object {
      "className": "org.example.JUnit4Test",
      "message": "",
      "result": "skipped",
      "stackTrace": undefined,
      "testName": "testIgnored",
    },
    Object {
      "className": "org.example.JUnit4Test",
      "message": "got: <false>, expected: is <true>",
      "result": "skipped",
      "stackTrace": "org.junit.AssumptionViolatedException: got: <false>, expected: is <true>
	at org.example.JUnit4Test.testSkipped(JUnit4Test.java:51)",
      "testName": "testSkipped",
    },
  ],
  "tests": 11,
}
`;

exports[`SurefireParser can parse a Surefire v2 report for JUnit 5 1`] = `
SurefireReport {
  "errors": 2,
  "failures": 4,
  "name": "org.example.JUnit5Test",
  "skipped": 4,
  "testCases": Array [
    Object {
      "className": "org.example.JUnit5Test",
      "message": "Assertion message ==> expected: <5> but was: <4>",
      "result": "failure",
      "stackTrace": "org.opentest4j.AssertionFailedError: Assertion message ==> expected: <5> but was: <4>
	at org.example.JUnit5Test.testAssertionFailureWithMessage(JUnit5Test.java:28)",
      "testName": "testAssertionFailureWithMessage",
    },
    Object {
      "className": "org.example.JUnit5Test",
      "message": "",
      "result": "error",
      "stackTrace": "java.lang.RuntimeException
	at org.example.JUnit5Test.testError(JUnit5Test.java:43)",
      "testName": "testError",
    },
    Object {
      "className": "org.example.JUnit5Test",
      "message": "",
      "result": "failure",
      "stackTrace": "org.opentest4j.AssertionFailedError: 
	at org.example.JUnit5Test.testFail(JUnit5Test.java:33)",
      "testName": "testFail",
    },
    Object {
      "className": "org.example.JUnit5Test",
      "message": "Disable message",
      "result": "skipped",
      "stackTrace": undefined,
      "testName": "testDisabledWithMessage",
    },
    Object {
      "className": "org.example.JUnit5Test",
      "message": "Error message",
      "result": "error",
      "stackTrace": "java.lang.RuntimeException: Error message
	at org.example.JUnit5Test.testErrorWithMessage(JUnit5Test.java:48)",
      "testName": "testErrorWithMessage",
    },
    Object {
      "className": "org.example.JUnit5Test",
      "message": "Assumption failed: Assumption message",
      "result": "skipped",
      "stackTrace": "org.opentest4j.TestAbortedException: Assumption failed: Assumption message
	at org.example.JUnit5Test.testSkippedWithMessage(JUnit5Test.java:58)",
      "testName": "testSkippedWithMessage",
    },
    Object {
      "className": "org.example.JUnit5Test",
      "message": "Failure message",
      "result": "failure",
      "stackTrace": "org.opentest4j.AssertionFailedError: Failure message
	at org.example.JUnit5Test.testFailWithMessage(JUnit5Test.java:38)",
      "testName": "testFailWithMessage",
    },
    Object {
      "className": "org.example.JUnit5Test",
      "message": "public void org.example.JUnit5Test.testDisabled() is @Disabled",
      "result": "skipped",
      "stackTrace": undefined,
      "testName": "testDisabled",
    },
    Object {
      "className": "org.example.JUnit5Test",
      "message": "expected: <5> but was: <4>",
      "result": "failure",
      "stackTrace": "org.opentest4j.AssertionFailedError: expected: <5> but was: <4>
	at org.example.JUnit5Test.testAssertionFailure(JUnit5Test.java:22)",
      "testName": "testAssertionFailure",
    },
    Object {
      "className": "org.example.JUnit5Test",
      "message": "Assumption failed: assumption is not true",
      "result": "skipped",
      "stackTrace": "org.opentest4j.TestAbortedException: Assumption failed: assumption is not true
	at org.example.JUnit5Test.testSkipped(JUnit5Test.java:53)",
      "testName": "testSkipped",
    },
  ],
  "tests": 11,
}
`;

exports[`SurefireParser can parse a Surefire v3 report for JUnit 4 1`] = `
SurefireReport {
  "errors": 2,
  "failures": 4,
  "name": "org.example.JUnit4Test",
  "skipped": 4,
  "testCases": Array [
    Object {
      "className": "org.example.JUnit4Test",
      "message": "",
      "result": "failure",
      "stackTrace": "java.lang.AssertionError: Assertion message expected:<5> but was:<4>
	at org.example.JUnit4Test.testAssertionFailureWithMessage(JUnit4Test.java:26)",
      "testName": "testAssertionFailureWithMessage",
    },
    Object {
      "className": "org.example.JUnit4Test",
      "message": "",
      "result": "error",
      "stackTrace": "java.lang.RuntimeException
	at org.example.JUnit4Test.testError(JUnit4Test.java:41)",
      "testName": "testError",
    },
    Object {
      "className": "org.example.JUnit4Test",
      "message": "",
      "result": "failure",
      "stackTrace": "java.lang.AssertionError
	at org.example.JUnit4Test.testFail(JUnit4Test.java:31)",
      "testName": "testFail",
    },
    Object {
      "className": "org.example.JUnit4Test",
      "message": "",
      "result": "error",
      "stackTrace": "java.lang.RuntimeException: Error message
	at org.example.JUnit4Test.testErrorWithMessage(JUnit4Test.java:46)",
      "testName": "testErrorWithMessage",
    },
    Object {
      "className": "org.example.JUnit4Test",
      "message": "Ignore message",
      "result": "skipped",
      "stackTrace": undefined,
      "testName": "testIgnoredWithMessage",
    },
    Object {
      "className": "org.example.JUnit4Test",
      "message": "",
      "result": "skipped",
      "stackTrace": "org.junit.AssumptionViolatedException: Assumption message
	at org.example.JUnit4Test.testSkippedWithMessage(JUnit4Test.java:56)",
      "testName": "testSkippedWithMessage",
    },
    Object {
      "className": "org.example.JUnit4Test",
      "message": "",
      "result": "failure",
      "stackTrace": "java.lang.AssertionError: Failure message
	at org.example.JUnit4Test.testFailWithMessage(JUnit4Test.java:36)",
      "testName": "testFailWithMessage",
    },
    Object {
      "className": "org.example.JUnit4Test",
      "message": "",
      "result": "failure",
      "stackTrace": "java.lang.AssertionError: expected:<5> but was:<4>
	at org.example.JUnit4Test.testAssertionFailure(JUnit4Test.java:20)",
      "testName": "testAssertionFailure",
    },
    Object {
      "className": "org.example.JUnit4Test",
      "message": "",
      "result": "skipped",
      "stackTrace": undefined,
      "testName": "testIgnored",
    },
    Object {
      "className": "org.example.JUnit4Test",
      "message": "",
      "result": "skipped",
      "stackTrace": "org.junit.AssumptionViolatedException: got: <false>, expected: is <true>
	at org.example.JUnit4Test.testSkipped(JUnit4Test.java:51)",
      "testName": "testSkipped",
    },
  ],
  "tests": 11,
}
`;

exports[`SurefireParser can parse a Surefire v3 report for JUnit 5 1`] = `
SurefireReport {
  "errors": 2,
  "failures": 4,
  "name": "org.example.JUnit5Test",
  "skipped": 4,
  "testCases": Array [
    Object {
      "className": "org.example.JUnit5Test",
      "message": "",
      "result": "failure",
      "stackTrace": "org.opentest4j.AssertionFailedError: Assertion message ==> expected: <5> but was: <4>
	at org.example.JUnit5Test.testAssertionFailureWithMessage(JUnit5Test.java:28)",
      "testName": "testAssertionFailureWithMessage",
    },
    Object {
      "className": "org.example.JUnit5Test",
      "message": "",
      "result": "error",
      "stackTrace": "java.lang.RuntimeException
	at org.example.JUnit5Test.testError(JUnit5Test.java:43)",
      "testName": "testError",
    },
    Object {
      "className": "org.example.JUnit5Test",
      "message": "",
      "result": "failure",
      "stackTrace": "org.opentest4j.AssertionFailedError: 
	at org.example.JUnit5Test.testFail(JUnit5Test.java:33)",
      "testName": "testFail",
    },
    Object {
      "className": "org.example.JUnit5Test",
      "message": "Disable message",
      "result": "skipped",
      "stackTrace": undefined,
      "testName": "testDisabledWithMessage",
    },
    Object {
      "className": "org.example.JUnit5Test",
      "message": "",
      "result": "error",
      "stackTrace": "java.lang.RuntimeException: Error message
	at org.example.JUnit5Test.testErrorWithMessage(JUnit5Test.java:48)",
      "testName": "testErrorWithMessage",
    },
    Object {
      "className": "org.example.JUnit5Test",
      "message": "",
      "result": "skipped",
      "stackTrace": "org.opentest4j.TestAbortedException: Assumption failed: Assumption message
	at org.example.JUnit5Test.testSkippedWithMessage(JUnit5Test.java:58)",
      "testName": "testSkippedWithMessage",
    },
    Object {
      "className": "org.example.JUnit5Test",
      "message": "",
      "result": "failure",
      "stackTrace": "org.opentest4j.AssertionFailedError: Failure message
	at org.example.JUnit5Test.testFailWithMessage(JUnit5Test.java:38)",
      "testName": "testFailWithMessage",
    },
    Object {
      "className": "org.example.JUnit5Test",
      "message": "public void org.example.JUnit5Test.testDisabled() is @Disabled",
      "result": "skipped",
      "stackTrace": undefined,
      "testName": "testDisabled",
    },
    Object {
      "className": "org.example.JUnit5Test",
      "message": "",
      "result": "failure",
      "stackTrace": "org.opentest4j.AssertionFailedError: expected: <5> but was: <4>
	at org.example.JUnit5Test.testAssertionFailure(JUnit5Test.java:22)",
      "testName": "testAssertionFailure",
    },
    Object {
      "className": "org.example.JUnit5Test",
      "message": "",
      "result": "skipped",
      "stackTrace": "org.opentest4j.TestAbortedException: Assumption failed: assumption is not true
	at org.example.JUnit5Test.testSkipped(JUnit5Test.java:53)",
      "testName": "testSkipped",
    },
  ],
  "tests": 11,
}
`;
